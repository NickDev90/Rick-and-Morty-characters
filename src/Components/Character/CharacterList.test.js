import { render, screen } from '@testing-library/react';
import CharacterList from './CharactersList.js';

const data = [{'name': 'John', 'status': 'alive', 'gender':'male', 'location': {'name': 'Earth'}}]
const emptyData = [];



describe('CharacterList component', ()=> {

    it('renders list of characters', () => {
      render(<CharacterList characters={data} key={data.index}/>);
      expect(screen.getByText('Earth')).toBeInTheDocument();
    });


    it('List snapshot', () => {
      const list = render(<CharacterList characters={data} key={data.index}/>);
      expect(list).toMatchSnapshot();
    });

    it('Empty list snapshot', () => {
      const list = render(<CharacterList characters={emptyData}/>);
      expect(list).toMatchSnapshot();
    });

  }
)

