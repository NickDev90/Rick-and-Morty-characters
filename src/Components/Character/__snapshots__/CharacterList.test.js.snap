// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterList component Empty list snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div />
    </div>
  </body>,
  "container": <div>
    <div />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CharacterList component List snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="description"
        >
          <span>
            Photo
          </span>
          <span
            class="nameField"
          >
            Name
          </span>
          <span>
            Gender
          </span>
          <span>
            Status
          </span>
          <span>
            Location
          </span>
        </div>
        <div
          class="heroInfo"
        >
          <span>
            <img
              class="heroPhoto"
              src="anon_photo.jpg"
            />
          </span>
          <span
            class="heroName"
          >
            John
          </span>
          <span>
            male
          </span>
          <span>
            alive
          </span>
          <span>
            Earth
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="description"
      >
        <span>
          Photo
        </span>
        <span
          class="nameField"
        >
          Name
        </span>
        <span>
          Gender
        </span>
        <span>
          Status
        </span>
        <span>
          Location
        </span>
      </div>
      <div
        class="heroInfo"
      >
        <span>
          <img
            class="heroPhoto"
            src="anon_photo.jpg"
          />
        </span>
        <span
          class="heroName"
        >
          John
        </span>
        <span>
          male
        </span>
        <span>
          alive
        </span>
        <span>
          Earth
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
